lib/osf-components/addon/components/addons-service/addon-account-setup/component.ts

```diff
@@ -197,20 +169,78 @@ export default class AddonAccountSetupComponent extends Component<Args> {
         }
     }
 
+    @task
+    @waitFor
+    async connectAccount() {
+        const { manager } = this.args;
+        const credentials = this.credentialsObject;
+        let apiBaseUrl;
+        if (this.showUrlField) {
+            apiBaseUrl = this.url;
+        } else if (this.showRepoOptions) {
+            apiBaseUrl = this.otherRepoSelected ? this.otherRepo : this.selectedRepo;
+        }
+        const accountCreationArgs = {
+            credentials,
+            apiBaseUrl,
+            displayName: this.displayName,
+            initiateOauth: this.useOauth,
+        };
+        try {
+            await taskFor(manager.connectAccount).unlinked().perform(accountCreationArgs);
```

Comment:
Need to add `unlinked` here since the `connectAccount` task on the UserAddonManager and Manager component will change the pageMode (and remove the `<AddonAccountSetup>` component from the page). More on `unlinked` here: https://ember-concurrency.com/api/Task.html